diff --git a/gym_mod/gym_mod/engine/genDisplay.py b/gym_mod/gym_mod/engine/genDisplay.py
index 8bb1b8b61aacf5efe5ea0c263aa1edfa73d40900..766baabe2c813e905e9f28856b466b91508cfb9d 100644
--- a/gym_mod/gym_mod/engine/genDisplay.py
+++ b/gym_mod/gym_mod/engine/genDisplay.py
@@ -10,32 +10,37 @@ def makeGif(numOfLife, name="", Type = "train", trunc = False):
     savePath = "display/"
 
     files = os.listdir(savePath)
     files = [os.path.join(savePath, f) for f in files]
     files.sort(key=lambda x: os.path.getmtime(x))
 
     if trunc == True:
         its = np.arange(numOfLife)
         itsChosen = np.random.choice(its-1, 30, replace=False)+1
         newFiles = []
         print(itsChosen)
 
         for i in files:
             for j in itsChosen:
                 digits = len(str(j))
                 print()
                 if i[:9+digits] == "display/{}_".format(j):
                     newFiles.append(i)
         files = newFiles
         print(files)
 
 
     for fil in tqdm(files):
         images.append(imageio.imread(fil))
     if Type == "train":
+        os.makedirs("metrics", exist_ok=True)
+        os.makedirs("gui/img", exist_ok=True)
+        os.makedirs("gui/build/img", exist_ok=True)
+        os.makedirs("gui/build/Debug/img", exist_ok=True)
         imageio.mimsave('metrics/model_train{}.gif'.format(name), images)
         imageio.mimsave('gui/img/model_train{}.gif'.format(name), images)
         imageio.mimsave('gui/build/img/model_train{}.gif'.format(name), images)
         imageio.mimsave('gui/build/Debug/img/model_train{}.gif'.format(name), images)
     elif Type == "val":
+        os.makedirs("val", exist_ok=True)
         imageio.mimsave('val/model_val.gif', images)
-    print("Generated gif")
\ No newline at end of file
+    print("Generated gif")

